### PR TITLE
Monorepo and releases

### LINKED ISSUE #31: Inconsistency in SDK methods
I just realized to delete a user, we need to pass `ID` but to delete a calendar we need to pass `DeleteCalendarInput` which contains one field for `ID`.

It's better to follow one pattern.

It can be there for all requests, delete/update/... and calendar/event/user/...

### PR DESCRIPTION
**Pull request by fmeringdal**
This PR should:
1. Setup structure for monorepo. That is NS server and clients (JS and Rust).
2. Setup a release process for automatic releases of server docker image and clients. In order to create a release just create a new tag on master branch with the format **vX.Y.Z**
3. Fixes #31 
4. CI / CD speedup with caches

### CODE REVIEW

**Comment by fmeringdal** [comment]
New draft for monorepo structure @omid . If you have any feedback let me know :)

File: scheduler/clients/rust/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-nettu_scheduler_api_structs = { path = "../api_structs", version = "0.3.1" }
-nettu_scheduler_domain = { path = "../domain", version = "0.2.1" }
+nettu_scheduler_api_structs = { path = "../../crates/api_structs", version = "0.3.1" }
+nettu_scheduler_domain = { path = "../../crates/domain", version = "0.2.1" }
```
Comment (omid) [inline scheduler/clients/rust/Cargo.toml:13]: I think we can remove all versions for internal deps from all `Cargo.toml`s

**Comment by omid** [comment]
If we need to do something specific to setup, check and test the Js library, we can put them in the `Makefile`.
Also we may need to have some more CI steps? I'm not sure

**Comment by fmeringdal** [comment]
I think it makes sense to have a new `Makefile` for each client, or what do you think?
I didnt start on CI / Release part yet, but will soon.

File: scheduler/clients/rust/Cargo.toml (hunk 1 above)
Comment (fmeringdal) [reply scheduler/clients/rust/Cargo.toml:13]: The version numbers are needed when publishing to crates.io.

This is the error you get if there is not a version:
```
$ cd scheduler/clients/rust
$ cargo publish
    Updating crates.io index
error: all dependencies must have a version specified when publishing.
dependency `nettu_scheduler_api_structs` does not specify a version
Note: The published dependency will use the version from crates.io,
the `path` specification will be removed from the dependency declaration.
```

Comment (omid) [reply scheduler/clients/rust/Cargo.toml:13]: ah, ok, didn't know that.

**Comment by omid** [comment]
I think one makefile in the root is better, we can have different commands there.

**Comment by fmeringdal** [comment]
I think the `Makefile` can be updated in a new PR since this PR is pretty large already :P 
Anyways I think everything is ready now regarding monorepo and releases. I put you on review @omid if you would like to have a look ;)

**Review by omid — APPROVED** [review]

**Comment by omid** [comment]
Sure. Thanks for the awesome changes 👍🏼

**Pull request merged**